src/main/java/org/ebyhr/trino/storage/operator/CsvPlugin.java

```diff
@@ -32,21 +34,24 @@ public class CsvPlugin
         implements FilePlugin
 {
     private final String delimiter;
+    private final CsvMapper mapper;
+    private final CsvSchema schema;
 
     public CsvPlugin(String delimiter)
     {
         this.delimiter = delimiter;
+        this.mapper = new CsvMapper();
+        this.mapper.enable(CsvParser.Feature.WRAP_AS_ARRAY).enable(CsvParser.Feature.TRIM_SPACES);
+        this.schema = CsvSchema.emptySchema().withColumnSeparator(this.delimiter.charAt(0));
     }
 
     @Override
     public List<StorageColumnHandle> getFields(String path, Function<String, InputStream> streamProvider)
     {
-        Splitter splitter = Splitter.on(delimiter).trimResults();
-
-        try (BufferedReader reader = new BufferedReader(new InputStreamReader(streamProvider.apply(path)))) {
-            List<String> fields = splitter.splitToList(reader.readLine());
-            return fields.stream()
-                    .map(field -> new StorageColumnHandle(field, VARCHAR))
+        try {
+            // Read the first line and use the values as column names
+            MappingIterator<List<String>> it = this.mapper.readerFor(List.class).with(schema).readValues(streamProvider.apply(path));
+            return it.next().stream().map(field -> new StorageColumnHandle(field, VARCHAR))
```

Comment:
Make it a tiny bit easier to read.
```suggestion
            MappingIterator<List<String>> it = this.mapper.readerFor(List.class).with(schema).readValues(streamProvider.apply(path));
            List<String> fields = it.next();
            return fields.stream()
                    .map(field -> new StorageColumnHandle(field, VARCHAR))
```

Reply:
Done https://github.com/snowlift/trino-storage/pull/244/commits/24754868b60bffc5a13ed0669a6532565ebc5c47

Reply:
Thanks, please squash all commits and we can merge this.